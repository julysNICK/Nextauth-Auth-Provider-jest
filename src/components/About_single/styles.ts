import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css``}
`;
export const Wrapper_details = styled.div`
  ${({ theme }) =>
    css`
      display: flex;
      flex-direction: column;
      gap: 1rem;
    `}
`;
export const Sections_name = styled.h1`
  ${({ theme }) =>
    css`
      text-align: center;
      color: #fff;
      text-decoration: underline;
    `}
`;
export const Details_movie = styled.span`
  ${({ theme }) =>
    css`
      color: #fff;
      font-size: 2rem;
      word-spacing: 0.2rem;
    `}
`;
export const Paragraph_single = styled.p`
  ${({ theme }) =>
    css`
      color: #fff;
      font-size: 2rem;
      word-spacing: 0.2rem;
    `}
`;
