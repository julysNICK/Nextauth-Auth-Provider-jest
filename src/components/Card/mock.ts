import { CardProps } from '.';
export default {
  imageCard:
    'https://i.pinimg.com/originals/db/2a/d1/db2ad13ff7535dabd9696d2f6ddb07ef.jpg',
  Levelcolor: 0,
  titleMovie: 'Velozes Furioso',
} as CardProps;
