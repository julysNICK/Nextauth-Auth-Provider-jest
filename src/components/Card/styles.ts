import styled, { css } from 'styled-components';
import Link from 'next/link';
type colorLevel = {
  color: string;
};
export const Wrapper = styled.div`
  ${({ theme }) =>
    css`
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: flex-end;
      margin: 10px;
      padding: 0.2rem;
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.9);

      width: 30rem;
      max-height: 300px;
      min-width: 100px;
      z-index: 1;
      &:hover {
        box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.9);
      }
    `}
`;

export const Wrapper_avaliation = styled.div`
  ${({ theme }) =>
    css`
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      margin-right: 15rem;
    `}
`;
export const Title = styled.h4`
  ${({ theme }) => css`
    margin: 0;
    color: rgb(255, 255, 255);
  `}
`;
export const Avaliation = styled.p`
  ${({ theme }) =>
    css`
      margin: 1rem 0 0 0;
      color: rgb(255, 255, 255);
    `}
`;

export const Title_link = styled.a`
  text-decoration: none;
  color: rgb(255, 255, 255);
`;

export const Image_card = styled.img`
  image-rendering: pixelated;
  max-height: 200px;
  width: 100%;
  object-fit: contain;
  margin-bottom: 15px;
`;
export const LINK_NEXT = styled(Link)`
  ${({ color }) =>
    css`
      text-decoration: none;
      color: rgb(255, 255, 255);
    `}
`;
