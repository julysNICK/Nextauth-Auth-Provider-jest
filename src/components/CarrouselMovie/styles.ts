import styled, { css } from 'styled-components';

import { ChevronLeft, ChevronRight } from '@styled-icons/boxicons-regular';
type SliderVisible = {
  Active?: boolean;
  lenght?: number;
};

export const Wrapper = styled.div<SliderVisible>`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: 1rem;
    display: flex;
    margin-top: 20rem;
    position: absolute;
    width: 320rem;
    overflow: hidden;
  `}
`;

export const Wrapper_image = styled.div<SliderVisible>`
  ${({ theme, lenght }) =>
    css`
      transform: translateX(${lenght === 0 ? 0 : lenght}%);
      transition: all 0.5s ease;
      overflow: hidden;
    `}
`;

export const Image_slider = styled.img`
  ${({ theme }) => css`
    position: relative;
    width: 50rem;
    height: 200px;
    border-radius: 10px;
  `}
`;

export const ArrowLeft = styled(ChevronLeft)`
  ${({ theme }) => css`
    height: 12rem;
    position: absolute;
    top: 50%;
    left: 0px;
    font-size: 3rem;
    color: #000;
    z-index: 10;
    cursor: pointer;
    user-select: none;
  `}
`;

export const ArrowRight = styled(ChevronRight)`
  ${({ theme }) => css`
    width: 10rem;
    position: absolute;
    top: 50%;
    right: 0px;
    font-size: 3rem;
    color: #000;
    z-index: 10;
    cursor: pointer;
    user-select: none;
  `}
`;
