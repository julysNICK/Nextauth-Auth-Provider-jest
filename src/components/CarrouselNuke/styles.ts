import styled, { css } from 'styled-components';
import Carousel from 'nuka-carousel';
import { ChevronLeft, ChevronRight } from '@styled-icons/boxicons-regular';
type SliderVisible = {
  Active?: boolean;
  lenght?: number;
};

export const Wrapper_container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  overflow: hidden;
`;

export const Wrapper_span = styled.div`
  display: flex;
  z-index: 6;
  position: absolute;
  align-self: flex-start;
  top: -1px;
  left: 1%;
`;
export const span = styled.span`
  color: #fff;
  font-size: 1.8rem;
`;
export const Wrapper_image = styled.div<SliderVisible>`
  position: relative;
  width: 50rem;
  height: 50rem;
  transition: all 0.5s ease;
`;

export const Image_slider = styled.img`
  border-radius: 10px;
`;
export const Carousel_wrape = styled(Carousel)`
  ${({ theme }) => css`
    position: absolute;
    margin-top: 2rem;
    .styles__Wrapper_image-sc-1lfhcjf-3 {
      height: 39rem;
      width: 23rem;
    }
    .slider-slide {
    }
  `}
`;
export const ArrowLeft = styled(ChevronLeft)`
  height: 12rem;

  top: 50%;
  left: 0px;
  font-size: 3rem;
  color: #fff;
  z-index: 10;
  cursor: pointer;
  user-select: none;
`;

export const ArrowRight = styled(ChevronRight)`
  height: 12rem;

  top: 50%;
  right: 0px;
  font-size: 3rem;
  color: #fff;
  z-index: 10;
  cursor: pointer;
  user-select: none;
  &:disabled {
    color: blue;
  }
`;
