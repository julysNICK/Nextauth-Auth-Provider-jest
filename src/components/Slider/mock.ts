export const SliderData = [
  {
    image:
      'https://www.publicitarioscriativos.com/wp-content/uploads/2018/01/16-mil-cartazes-de-filmes-e-series-em-hd-sem-textos-3.jpg',
  },
  {
    image:
      'https://www.publicitarioscriativos.com/wp-content/uploads/2018/01/16-mil-cartazes-de-filmes-e-series-em-hd-sem-textos-4.jpg',
  },
  {
    image:
      'https://www.publicitarioscriativos.com/wp-content/uploads/2018/01/16-mil-cartazes-de-filmes-e-series-em-hd-sem-textos-1.jpg',
  },
  {
    image:
      'https://images.unsplash.com/photo-1475189778702-5ec9941484ae?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1351&q=80',
  },
  {
    image:
      'https://images.unsplash.com/photo-1503177119275-0aa32b3a9368?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1350&q=80',
  },
  {
    image:
      'https://images.unsplash.com/photo-1503177119275-0aa32b3a9368?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1350&q=80',
  },
];
