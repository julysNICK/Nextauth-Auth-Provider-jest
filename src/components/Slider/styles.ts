import styled, { css } from 'styled-components';

import { ChevronLeft, ChevronRight } from '@styled-icons/boxicons-regular';
type SliderVisible = {
  Active?: boolean;
};
const SliderActive = () => css`
  opacity: 1;
  transition-duration: 1s;
  transform: scale(1.08);
`;
export const Wrapper = styled.div`
  ${({ theme }) =>
    css`
      display: flex;
      position: relative;
      height: 100vh;
      justify-content: center;
      align-items: center;
    `}
`;

export const Wrapper_image = styled.div<SliderVisible>`
  ${({ theme, Active }) => css`
    opacity: 0;
    transition-duration: 1s ease;
    ${Active && SliderActive()}
  `}
`;

export const Image_slider = styled.img`
  ${({ theme }) => css`
    width: 1000px;
    height: 600px;
    border-radius: 10px;
  `}
`;

export const ArrowLeft = styled(ChevronLeft)`
  ${({ theme }) => css`
    height: 12rem;
    position: absolute;
    top: 50%;
    left: 32px;
    font-size: 3rem;
    color: #000;
    z-index: 10;
    cursor: pointer;
    user-select: none;
  `}
`;

export const ArrowRight = styled(ChevronRight)`
  ${({ theme }) => css`
    width: 10rem;
    position: absolute;
    top: 50%;
    right: 32px;
    font-size: 3rem;
    color: #000;
    z-index: 10;
    cursor: pointer;
    user-select: none;
  `}
`;
