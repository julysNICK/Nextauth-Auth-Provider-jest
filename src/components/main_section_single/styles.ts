import Carousel from 'nuka-carousel';
import styled, { css } from 'styled-components';
type SliderVisible = {
  Active?: boolean;
};

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 30vw;
    height: 40vh;

    margin: 0 auto;
  `}
`;

export const Title_single = styled.h1`
  color: #fff;
  margin: 0;
  word-wrap: break-word;
`;
export const Image = styled.img<SliderVisible>`
  image-rendering: pixelated;
  object-fit: contain;

  object-fit: contain;
`;
export const Carousel_wrape = styled(Carousel)`
  ${({ theme }) => css``}
`;
