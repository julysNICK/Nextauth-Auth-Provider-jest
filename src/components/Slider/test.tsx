import { fireEvent, screen } from '@testing-library/react';

import { renderTheme } from '../../styles/render-theme';
import { SliderData } from './mock';
import { Slider, SliderProps } from '.';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
const props: SliderProps = {
  slides: [],
};
export const SliderDataa = [
  {
    image:
      'https://images.unsplash.com/photo-1546768292-fb12f6c92568?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1350&q=80',
  },
];
describe('<Slider />', () => {
  it('should render component son null', () => {
    const { container } = renderTheme(<Slider slides={undefined} />);

    expect(container.nextSibling).toBeNull();
  });

  it('should render component with img', () => {
    renderTheme(<Slider slides={SliderData} />);

    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('should render component clicked button prev to pass the image', () => {
    renderTheme(<Slider slides={SliderData} />);

    const button = screen.getByTestId('prev_slide');

    fireEvent.click(button);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('should render component clicked button next to pass the image', () => {
    renderTheme(<Slider slides={SliderData} />);

    const button = screen.getByTestId('next_slide');

    fireEvent.click(button);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('should render', () => {
    renderTheme(<Slider slides={SliderDataa} />);

    const button = screen.getByTestId('next_slide');

    fireEvent.click(button);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('should render last img', () => {
    const { rerender } = renderTheme(<Slider slides={SliderData} />);

    const button = screen.getByTestId('prev_slide');

    fireEvent.click(button);
    rerender(
      <ThemeProvider theme={theme}>
        <Slider slides={SliderData} />
      </ThemeProvider>,
    );
    const button1 = screen.getByTestId('prev_slide');

    fireEvent.click(button1);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('should render component snapshot', () => {
    const { container } = renderTheme(<Slider slides={SliderData} />);

    expect(container).toMatchSnapshot();
  });
});
