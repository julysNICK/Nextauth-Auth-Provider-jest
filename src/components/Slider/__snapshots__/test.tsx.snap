// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render component snapshot 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  opacity: 0;
  -webkit-transition-duration: 1s ease;
  transition-duration: 1s ease;
  opacity: 1;
  -webkit-transition-duration: 1s;
  transition-duration: 1s;
  -webkit-transform: scale(1.08);
  -ms-transform: scale(1.08);
  transform: scale(1.08);
}

.c6 {
  opacity: 0;
  -webkit-transition-duration: 1s ease;
  transition-duration: 1s ease;
}

.c5 {
  width: 1000px;
  height: 600px;
  border-radius: 10px;
}

.c3 {
  height: 12rem;
  position: absolute;
  top: 50%;
  left: 32px;
  font-size: 3rem;
  color: #000;
  z-index: 10;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  width: 10rem;
  position: absolute;
  top: 50%;
  right: 32px;
  font-size: 3rem;
  color: #000;
  z-index: 10;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="c1 c2"
      data-testid="prev_slide"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.707 17.707L16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c1 c3"
      data-testid="next_slide"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.293 6.293L7.586 12l5.707 5.707 1.414-1.414L10.414 12l4.293-4.293z"
      />
    </svg>
    <div
      class="c4"
    >
      <img
        alt="travel image"
        class="c5"
        src="https://www.publicitarioscriativos.com/wp-content/uploads/2018/01/16-mil-cartazes-de-filmes-e-series-em-hd-sem-textos-3.jpg"
      />
    </div>
    <div
      class="c6"
    />
    <div
      class="c6"
    />
    <div
      class="c6"
    />
    <div
      class="c6"
    />
    <div
      class="c6"
    />
  </div>
</div>
`;
